hc2/headers/types/code_object_bundle.hpp

```diff
@@ -133,28 +133,26 @@ namespace hc2
     inline
     hsa_isa_t triple_to_hsa_isa(const std::string& triple)
     {
-        static constexpr const char gfxip[] = "hcc-amdgcn--amdhsa-gfx";
-
-        hsa_isa_t r = {};
-
-        auto it = std::find_if(
-            triple.cbegin(),
-            triple.cend(),
-            [](char x) { return std::isdigit(x); });
-        if (std::equal(triple.cbegin(), it, gfxip)) {
-            std::string tmp = "AMD:AMDGPU";
-            while (it != triple.cend()) {
-                tmp.push_back(':');
-                tmp.push_back(*it++);
-            }
+      static constexpr const char OffloadKind[] = "hcc-";
+      hsa_isa_t Isa = {0};
 
-            throwing_hsa_result_check(
-                hsa_isa_from_name(tmp.c_str(), &r),
-                __FILE__,
-                __func__,
-                __LINE__);
-        }
+      if ((triple.size() < sizeof(OffloadKind) - 1) || !std::equal(triple.c_str(),  triple.c_str() + sizeof(OffloadKind) - 1, triple.c_str())) {
+        return Isa;
+      }
+
+      std::string validatedTriple = triple.substr(sizeof(OffloadKind) - 1);
+      static constexpr const char oldPrefix[] = "amdgcn--amdhsa-gfx";
+      static constexpr const char newPrefix[] = "amdgcn-amd-amdhsa--gfx";
+
+      if ((triple.size() >= sizeof(oldPrefix) - 1) && std::equal(oldPrefix, oldPrefix + sizeof(oldPrefix) - 1, triple.c_str())) {
+        // Support backwards compatibility with old naming.
+        validatedTriple = newPrefix + triple.substr(sizeof(oldPrefix) - 1);
```

Comment:
same here, should replace `triple` with `validatedTriple`

Reply:
Also do same to HIP version.